service/src/main/java/greencity/service/ubs/UBSClientServiceImpl.java

```diff
@@ -359,6 +372,105 @@ public OrderWithAddressesResponseDto saveCurrentAddressForOrder(OrderAddressDtoR
         return findAllAddressesForCurrentOrder(uuid);
     }
 
+    private OrderAddressDtoRequest geoCodeSearchingRequest(String searchRequest) {
+        List<Locale> locales = List.of(new Locale("uk"), new Locale("en"));
+        GeoApiContext context = new GeoApiContext.Builder().apiKey(googleApiKey).build();
+        List<GeocodingResult> geocodingResults = new ArrayList<>();
+
+        locales.forEach(locale -> {
+            try {
+                GeocodingResult[] results = GeocodingApi.newRequest(context)
+                    .address(searchRequest).language(locale.getLanguage()).await();
+                Collections.addAll(geocodingResults, results);
+            } catch (IOException | InterruptedException | ApiException e) {
+                throw new RuntimeException(e.getMessage());
+            }
+        });
+        return getLocationDto(geocodingResults);
+    }
+
+    private OrderAddressDtoRequest getLocationDto(List<GeocodingResult> geocodingResults) {
+        OrderAddressDtoRequest orderAddressDtoRequest = new OrderAddressDtoRequest();
+        GeocodingResult ukrLang = geocodingResults.get(0);
+        GeocodingResult engLang = geocodingResults.get(1);
+
+        Arrays.stream(ukrLang.addressComponents).forEach(addressComponent -> {
+            Arrays.stream(addressComponent.types).filter(type -> type.equals(AddressComponentType.LOCALITY))
+                .forEach(type -> orderAddressDtoRequest.setCity(addressComponent.longName));
+        });
+        Arrays.stream(engLang.addressComponents).forEach(addressComponent -> {
+            Arrays.stream(addressComponent.types).filter(type -> type.equals(AddressComponentType.LOCALITY))
+                .forEach(type -> orderAddressDtoRequest.setCityEn(addressComponent.longName));
+        });
+
+        Arrays.stream(ukrLang.addressComponents).forEach(addressComponent -> {
+            Arrays.stream(addressComponent.types).filter(type -> type.equals(AddressComponentType.STREET_NUMBER))
+                .forEach(type -> orderAddressDtoRequest.setHouseNumber(addressComponent.longName));
+        });
+
+        Arrays.stream(engLang.addressComponents).forEach(addressComponent -> {
```

Comment:
Arrays.stream .... - is a copy/paste. 
You can define static map, where you'll have key: AddressComponentType, and value -  Consumer<String>, right?

Reply:
yep, it's copy-paste but I don't  see any reason to create the map, because in stream we're setting the value to dto;
So it will be the same copy-paste but with the consumer's and map that not be used.